src/Controller/ApiV1Controller.php

```diff
@@ -0,0 +1,211 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * PHP version 5
+ *
+ * @category PHP
+ * @package  Tatoeba
+ * @author   HO Ngoc Phuong Trang <tranglich@gmail.com>
+ * @license  Affero General Public License
+ * @link     https://tatoeba.org
+ */
+namespace App\Controller;
+
+use App\Controller\AppController;
+use App\Form\SentencesSearchForm;
+use App\Model\CurrentUser;
+use App\Model\Table\SentencesTable;
+use App\Lib\LanguagesLib;
+use App\Lib\SphinxClient;
+use App\Lib\Licenses;
+use Cake\Core\Configure;
+use Cake\Database\Expression\QueryExpression;
+use Cake\Event\Event;
+use Cake\Routing\Router;
+use Cake\Utility\Hash;
+use Cake\View\ViewBuilder;
+use Exception;
+
+/**
+ * Controller for Api.
+ *
+ * @category Sentences
+ * @package  Controllers
+ * @author   HO Ngoc Phuong Trang <tranglich@gmail.com>
+ * @license  Affero General Public License
+ * @link     https://tatoeba.org
+ */
+class ApiV1Controller extends AppController
+{
+    public $name = 'Sentences';
+    public $components = array (
+        'LanguageDetection',
+        'CommonSentence',
+        'Permissions',
+    );
+    public $helpers = array(
+        'Sentences',
+        'Menu',
+        'Lists',
+        'SentenceButtons',
+        'Html',
+        'Logs',
+        'Pagination',
+        'Comments',
+        'Languages',
+        'CommonModules'
+    );
+
+    public $uses = array(
+        'Audio',
+        'Sentence',
+        'SentencesSentencesLists',
+        'SentencesList',
+        'User',
+        'UsersLanguages',
+        'Tag',
+        'UsersSentences',
+        'Vocabulary'
+    );
+
+    public function initialize() {
+        parent::initialize();
+
+        $params = $this->request->params;
+        $noCsrfActions = ['edit_sentence', 'change_language'];
+        if (in_array($params['action'], $noCsrfActions)) {
+            $this->components()->unload('Csrf');
+        }
+    }
+
+    /**
+     * Show Json of specified sentence id.
+     * 
+     * @param mixed $id Id of the sentence.
+     * 
+     * @return void
+     */
+    public function sentence($id = null) {
+        // We will handle the output directly, so as not to display the header and footer
+        $this->disableAutoRender();
+
+        if (is_numeric($id)) {
+            // Retrieve the sentence
+            $sentence = $this->Sentences->getSentenceWith($id, [
+                'translations' => true,
+                'sentenceDetails' => true
+            ]);
+
+
+            return $this->getResponse()
+                ->withStringBody($sentence);
+        }
+    }
+
+    /**
+     * Show in Json format a list of sentence ids which matches a given search.
+     * 
+     * Note that this method is basically a copy/paste of the SentencesController.php 
+     * search() method. Only the minimum changes necessary to deliver an array of ids
+     * have been made. Therefore, there may be some redundant code below.
```

Comment:
Please remove the redundant code instead of writing a comment like this. Alternatively, you can remove the search method from your PR and submit it another time.